tardis/visualization/widgets/shell_info.py

```diff
@@ -232,263 +238,201 @@ class ShellInfoWidget:
-                [
-                    self.shells_table,
-                    self.element_count_table,
-                    self.ion_count_table,
-                    self.level_count_table,
-                ],
-                layout=ipw.Layout(**tables_container_layout),
-            )
-            self.shells_table.change_selection([1])
-
-            # Notes text explaining how to interpret tables widgets' data
-            text = ipw.HTML(
-                "<b>Frac. Ab.</b> denotes <i>Fractional Abundances</i> (i.e all "
-                "values sum to 1)<br><b>W</b> denotes <i>Dilution Factor</i> and "
-                "<b>Rad. Temp.</b> is <i>Radiative Temperature (in K)</i>"
-            )
+            self.element_count_table.value = pd.DataFrame(columns=["Element", "Frac. Ab."])
+            self.element_title.object = "### Elements (No Shell Selected)"
+            # self.element_count_table.selection = []
+            self.ion_count_table.value = pd.DataFrame(columns=["Species", "Frac. Ab."])
+            self.ion_title.object = "### Ions (No Selection)"
+    
+            self.level_count_table.value = pd.DataFrame(columns=["Level", "Frac. Ab."])
+            self.level_title.object = "### Levels (No Selection)"
+            self.update_ion_count_table(None)
+
+            
+    def update_ion_count_table(self, event):
+        """Update ion table based on element selection"""
+        if event and self.shells_table.selection:
+            selected_rows = event.new
+            shell_num = self.shells_table.selection[0] + 1
+            if selected_rows:
+                atomic_num = self.element_count_table.value.index[selected_rows[0]]
+                ion_df = self.data.ion_count(atomic_num, shell_num)
+                self.ion_count_table.value = ion_df
+                self.ion_title.object = f"### Ions (Z={atomic_num}, Shell {shell_num})"
+                self.ion_count_table.widths = {
+                    "Species": 140,
+                    f"Frac. Ab. (Z={atomic_num})": 140,
+                }
+                self._apply_tabulator_styles(self.ion_count_table)
+
+                if not ion_df.empty:
+                    self.ion_count_table.selection = [0]
+                else:
+                    self.ion_count_table.selection = []
+                    self.update_level_count_table(None)
+            else:
+                self.ion_count_table.value = pd.DataFrame(columns=["Species", "Frac. Ab."])
+                self.ion_title.object = "### Ions (No Element Selected)"
+                self.ion_count_table.selection = []
+                self.update_level_count_table(None)
+        else:
+            self.ion_count_table.value = pd.DataFrame(columns=["Species", "Frac. Ab."])
+            self.ion_title.object = "### Ions (No Selection)"
+            self.ion_count_table.selection = []
+            self.update_level_count_table(None)
+
+    def update_level_count_table(self, event):
```

Comment:
There's redundant logic in this and above functions that can be abstracted in a function and then called here